src/conn/conn_tiered.c

```diff
@@ -175,32 +175,49 @@ __tiered_manager_config(WT_SESSION_IMPL *session, const char **cfg, bool *runp)
  *     Parse and setup the storage server options.
  */
 static int
-__tiered_config(WT_SESSION_IMPL *session, const char **cfg, bool *runp)
+__tiered_config(WT_SESSION_IMPL *session, const char **cfg, bool *runp, bool reconfig)
 {
-    WT_CONFIG_ITEM cval;
+    WT_CONFIG_ITEM bucket, cval;
     WT_CONNECTION_IMPL *conn;
-    bool enabled;
+    WT_DECL_RET;
 
     conn = S2C(session);
 
-    WT_RET(__wt_config_gets(session, cfg, "tiered_storage.enabled", &cval));
-    enabled = cval.val != 0;
+    if (!reconfig) {
+        WT_RET(__wt_config_gets(session, cfg, "tiered_storage.name", &cval));
+        WT_RET(__wt_config_gets(session, cfg, "tiered_storage.bucket", &bucket));
+        WT_RET(__wt_tiered_bucket_config(session, &cval, &bucket, &conn->bstorage));
+    }
+    /* If the connection is not set up for tiered storage there is nothing more to do. */
+    if (conn->bstorage == NULL)
+        return (0);
 
-    if (enabled)
-        FLD_SET(conn->tiered_flags, WT_CONN_TIERED_ENABLED);
-    else
-        FLD_CLR(conn->tiered_flags, WT_CONN_TIERED_ENABLED);
+    WT_RET(__wt_config_gets(session, cfg, "tiered_storage.local_retention", &cval));
+    conn->bstorage->retain_secs = (uint64_t)cval.val;
+    WT_STAT_CONN_SET(session, tiered_retention, conn->bstorage->retain_secs);
+
+    WT_RET(__wt_config_gets(session, cfg, "tiered_storage.object_target_size", &cval));
+    WT_ASSERT(session, conn->bstorage != NULL);
+    conn->bstorage->object_size = (uint64_t)cval.val;
```

Comment:
There's a little bit of shared code here with where bstorage is set in `__btree_config_tiered`.  We may well add more configuration like this to control bucket storage cache, etc., so maybe good to have a shared function?

Reply:
fixed